File: php/shapes/Quadrant.php
<?php

enum Quadrant
{
    case FIRST;
    case SECOND;
    case THIRD;
    case FORTH;

    public function getX(): int
    {
        return match ($this) {
            self::FIRST, self::FORTH => 1,
            self::SECOND, self::THIRD => -1,
        };
    }

    public function getY(): int
    {
        return match ($this) {
            self::FIRST, self::SECOND => 1,
            self::THIRD, self::FORTH => -1,
        };
    }
}

File: src/main/java/com/moleus/web/service/areaCheck/quadrant/Quadrant.java
package com.moleus.web.service.areaCheck.quadrant;

public enum Quadrant {
    FIRST(1, 1),
    SECOND(-1, 1),
    THIRD(-1, -1),
    FORTH(1, -1);

    public final int x;
    public final int y;

    Quadrant(int x, int y) {
        this.x = x;
        this.y = y;
    }
}
